### ISSUE TITLE
Language Server crashes since latest update of C/C++ 0.21.0

### ISSUE THREAD

**Issue by ThatGuyMike7** [microsoft/vscode-cpptools#3086]
C/C++ package says that the language server crashed when you open a source file that includes and uses functions from a custom shared library project.

Error message (displayed in bottom-right corner): **The language server crashed 5 times in the last 3 minutes. It will not be restarted.**

Error message (displayed in "Output"): 
`[Error - 3:10:00 PM] Connection to server got closed. Server will not be restarted.
[Error - 3:10:00 PM] Request textDocument/codeAction failed.
Error: Connection got disposed.
	at Object.dispose (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\main.js:876:25)
	at Object.dispose (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-languageclient\lib\client.js:71:35)
	at LanguageClient.handleConnectionClosed (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-languageclient\lib\client.js:2150:42)
	at LanguageClient.handleConnectionClosed (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-languageclient\lib\main.js:150:15)
	at closeHandler (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-languageclient\lib\client.js:2137:18)
	at CallbackList.invoke (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at closeHandler (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\main.js:226:26)
	at CallbackList.invoke (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\events.js:62:39)
	at Emitter.fire (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\events.js:120:36)
	at StreamMessageReader.fireClose (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\messageReader.js:111:27)
	at Socket.AbstractMessageReader.listen.callback.readable.on (C:\Users\Mike\.vscode\extensions\ms-vscode.cpptools-0.21.0\node_modules\vscode-jsonrpc\lib\messageReader.js:151:46)
	at emitOne (events.js:121:20)
	at Socket.emit (events.js:211:7)
	at Pipe.Socket._destroy.cb._handle.close [as _onclose] (net.js:554:12)`

- VSCode Version: 1.30.2
- OS Version: Windows_NT x64 10.0.17134
- C/C++ Extension Version: 0.21.0
- Other perhaps relevant packages I used: CMake Tools, Project Manager, Include Autocomplete

Steps to Reproduce:

1. Download C/C++ and CMake Tools packages
2. Set up a CMake Project with multiple translation units:
    --> One shared library (.dll)
    --> One executable (.exe)
3. Include and use functions from the **shared library** project in the **executable** project and see how the langauge server crashes when you open that specific source file (for me it was main.cpp)

Does this issue occur when all extensions are disabled?: No

**Note:** When I roll back to C/C++ Version 0.20.1, the issue is gone!

**Comment by sean-mcmanus** [comment]
Can you provide a sample repro project? Can you attach a debugger to the Microsoft.VSCode.CPP.Extension.exe process before it crashes and provide a call stack?

**Comment by ThatGuyMike7** [comment]
`Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175D270.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175D270.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175D270.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175CCC0.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175C27C.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175EFDC.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175EFDC.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E910.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: antlr::NoViableAltException at memory location 0x0175E4D4.
Exception thrown at 0x75051812 in Microsoft.VSCode.CPP.Extension.exe: Microsoft C++ exception: [rethrow] at memory location 0x00000000.`

This is the output in the Debug Console when I attach a debugger to the process. Not sure how much this helps you, I will try making a sample project.

The exceptions were thrown all the time, since I first attached the Debugger. At the end of the messages, I switched to the `main.cpp` tab.
